docs/sources/configure-client/grafana-agent/sampling.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Sampling scrape targets"
+menuTitle: "Sampling targets"
+description: "Sampling scrape targets with the Grafana Agent"
+weight: 30
+---
+
+# Sampling scrape targets
+
+It's not uncommon for an application to have many instances deployed. While Pyroscope is designed specifically to handle large amounts of profiling data, you may have other considerations which demand that only a subset of the application's instances should be scraped. For example, the sheer volume of profiling data your application generates may make it unreasonable to profile every instance or you might be targeting cost-reduction. Whatever the case may be, it's possible to sample scrape targets through configuration of the Grafana Agent.
+
+## Prerequisites
+
+Before you begin, make sure you understand how to [configure the Grafana Agent]({{< relref "." >}}) to scrape targets and are familiar with the Grafana Agent [component configuration language]({{< relref "/docs/agent/latest/flow/config-language/components" >}}).
+
+## Configuration
+
+The `hashmod` action and the `modulus` argument are used in conjunction to enable sampling behavior. To read further on these concepts, see [rule block documentation]({{< relref "/docs/agent/latest/flow/reference/components/discovery.relabel#rule-block" >}}). In short, `hashmod` will perform an MD5 hash on the source labels and `modulus` will perform a modulus operation on the output.
+
+Together, these can be leveraged to shard targets into shards. One or more shards can be selected to be the sampled targets to be scraped. Here is an example:
```

Comment:
I'm not sure I understand the phrase "share targets into shards." If you have better wording, please feel free to edit this suggestion to update. 
```suggestion
Together, they can be leveraged to shard targets into shards. One or more shards can be selected to be the sampled targets to be scraped. Here is an example:
```

Reply:
```suggestion
Together, these can be leveraged to divide targets into shards. Once divided, you can configure what percentage of the shards should be scraped. Here is an example:
```

Reply:
In In https://github.com/grafana/pyroscope/pull/2854/commits/18daf624396b43aa46699122228e2f2edee0e7e1, I also reworked this part. I went into a little more detail about how `hashmod` will shard the labels and `keep` will select shards.

It might still need some work.